scripts/rename_classes.py

```
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+"""
+Class Name Renaming Script
+
+This script automatically renames class names in a codebase according to a predefined mapping.
+It handles both class definitions and usages throughout the code.
+"""
+
+import os
+import re
+import sys
+from typing import Dict, List, Tuple
```

Ruff (0.8.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)